pkg/server/plugin/upstreamca/awssecret/awssecret.go

```diff
@@ -0,0 +1,238 @@
+
+	if block == nil {
+		return nil, nil, errors.New("invalid key format")
+	}
+
+	if len(rest) > 0 {
+		return nil, nil, errors.New("invalid key format: too many keys")
+	}
+
+	key, err := x509.ParseECPrivateKey(block.Bytes)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	certPEMstr, err := readARN(sm, config.CertFileARN)
+
+	if certPEMstr == nil || err != nil {
+		return nil, nil, fmt.Errorf("unable to read %s: %s", config.CertFileARN, err)
+	}
+
+	certPEM := []byte(*certPEMstr)
+
+	block, rest = pem.Decode(certPEM)
+
+	if block == nil {
+		return nil, nil, errors.New("invalid cert format")
+	}
+
+	if len(rest) > 0 {
+		return nil, nil, errors.New("invalid cert format: too many certs")
+	}
+
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Validate cert matches private key
+	matched, err := x509util.CertificateMatchesPrivateKey(cert, key)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if !matched {
+		return nil, nil, errors.New("certificate and private key does not match")
+	}
+
+	return key, cert, nil
+}
+
+func (m *awssecretPlugin) configValidation(ctx context.Context, req *spi.ConfigureRequest) (*AWSSecretConfiguration, error) {
+	// Parse HCL config payload into config struct
+
+	config := new(AWSSecretConfiguration)
+
+	if err := hcl.Decode(&config, req.Configuration); err != nil {
+		return nil, err
+	}
+
+	if req.GlobalConfig == nil {
+		return nil, errors.New("global configuration is required")
+	}
+
+	if req.GlobalConfig.TrustDomain == "" {
+		return nil, errors.New("trust_domain is required")
+	}
+
+	// Set defaults from the environment
+	if config.SecurityToken == "" {
+		config.SecurityToken = m.hooks.getenv("AWS_SESSION_TOKEN")
+	}
+
+	switch {
+	case config.CertFileARN != "" && config.KeyFileARN != "":
+	case config.CertFileARN != "" && config.KeyFileARN == "":
+		return nil, errors.New("configuration missing key ARN")
+	case config.CertFileARN == "" && config.KeyFileARN != "":
+		return nil, errors.New("configuration missing cert ARN")
+	case config.CertFileARN == "" && config.KeyFileARN == "":
+		return nil, errors.New("configuration missing both cert ARN and key ARN")
+	}
+
+	return config, nil
+}
+
+func (m *awssecretPlugin) Configure(ctx context.Context, req *spi.ConfigureRequest) (*spi.ConfigureResponse, error) {
+	var (
+		sm  secretsmanageriface.SecretsManagerAPI
+		err error
+	)
+
+	config, err := m.configValidation(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	// set the AWS configuration and reset clients
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	m.config = config
+
+	// Exposed for testing
+	if config.UseFakeSecretsManager == "yes" {
+		sm, err = newFakeSecretsManagerClient()
+	} else {
+		sm, err = newSecretsManagerClient(config, config.Region)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	key, cert, err := fetchFromSecretsManager(config, sm)
+	if err != nil {
+		return nil, err
+	}
+
+	ttl, err := time.ParseDuration(config.TTL)
```

Comment:
this should happen within `configValidation`. if you don't want to return it from config validation, i'd recommend adding a `ttl time.Duration` field to the Config (unexported) and populate it within configValidation.

Reply:
`time.Duration` in hcl only supports nanoseconds which seems unexpected, adding a return value instead.